### ISSUE TITLE
spring-mvc config.json server should have an option for configPackage

### ISSUE THREAD

**Issue by cosbor11** [swagger-api/swagger-codegen#905]
see title

**Comment by wing328** [comment]
Are you referring to the following options ?

```
swagger-codegen|master ⇒ java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar config-help -l spring-mvc

CONFIG OPTIONS
	modelPackage
	    package for generated models

	apiPackage
	    package for generated api classes

	invokerPackage
	    root package for generated code

	groupId
	    groupId in generated pom.xml

	artifactId
	    artifactId in generated pom.xml

	artifactVersion
	    artifact version in generated pom.xml

	sourceFolder
	    source folder for generated code
```

**Comment by cosbor11** [comment]
Yes

**Comment by wing328** [comment]
would you please elaborate a bit on the configPackage ? What does it do/change exactly ?

I checked in [JavaClientCodegen.java] (https://github.com/swagger-api/swagger-codegen/blob/develop_2.0/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java) but couldn't find any configPackage

**Comment by cosbor11** [comment]
Sorry, i wasnt very clear...It's the output folder name.
"io.swagger.configuration" would like to be able to rename to
com.mycompany.api.config. similarly you can do this to the other folders.

**Comment by herveDarritchon** [comment]
In fact, it seems that the Swagger CodeGen is not able to override the configPackage variable set in the code even if you put a configPackage in the config json file.

Look at my example :

If I use the standard example : PetStore

If I run the script `./bin/spring-mvc-petstore-server.sh`with a modification in the run command line such as `ags="$@ generate -t modules/swagger-codegen/src/main/resources/JavaSpringMVC -i modules/swagger-codegen/src/test/resources/2_0/petstore.json -c /Users/herveDarritchonworkspace/ms-admin-data-api/src/main/swagger/config.json -l spring-mvc -o samples/server/petstore/spring-mvc"`

Please find the config.json file content :
```
{
  "groupId":"com.orange.dsifapi.microservices",
  "artifactId":"admindata",
  "artifactVersion":"0.0.1",
  "modelPackage":"com.orange.dsifapi.microservice.admindata.domain",
  "apiPackage":"com.orange.dsifapi.microservice.admindata.api",
  "invokerPackage":"com.orange.dsifapi.microservice.admindata.invoker",
  "configPackage":"com.orange.dsifapi.microservice.admindata.configuration",
  "title":"API d'Admin des données d'Habilitation."
}
```

The result of the command is quite clear all the variables are overrided except the configPackage.

```
reading config from /Users/throdo/Programmation/Workspaces/Orange/API/ProtoHabilitation/ms-admin-data-api/src/main/swagger/config.json
reading from modules/swagger-codegen/src/test/resources/2_0/petstore.json
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/domain/User.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/domain/Category.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/domain/Pet.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/domain/Tag.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/domain/Order.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/StoreApi.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/UserApi.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/PetApi.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/pom.xml
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/README.md
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiException.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiOriginFilter.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiResponseMessage.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/com/orange/dsifapi/microservice/admindata/api/NotFoundException.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/io/swagger/configuration/SwaggerConfig.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/io/swagger/configuration/WebApplication.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/io/swagger/configuration/WebMvcConfiguration.java
writing file /Users/throdo/Programmation/Workspaces/Orange/Swagger/swagger-codegen/samples/server/petstore/spring-mvc/src/main/java/io/swagger/configuration/SwaggerUiConfiguration.java
```

It looks like the configPackage cannot be overridden by configuration.

May be it's because it is typical to Spring MVC, you don't have such variable in the Jersey template for example.

If I have little time, I will try to analyze more deeply in the code of the codegen to find the bug and may be make a PR except someone is already trying to fix it.

I think it's a bug ;)

**Comment by fehguy** [comment]
@wing328 is this something that's not supported in the config file reader?  I belive it should be.

**Comment by wing328** [comment]
It's fixed by #1017. Confirmed below:
```
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiException.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiOriginFilter.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/api/ApiResponseMessage.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/api/NotFoundException.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/configuration/SwaggerConfig.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/configuration/WebApplication.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/configuration/WebMvcConfiguration.java
writing file /var/tmp/springmvc/test/src/main/java/com/orange/dsifapi/microservice/admindata/configuration/SwaggerUiConfiguration.java
```
